src/Components/Footer/Footer.tsx

```diff
@@ -0,0 +1,101 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import { deleteTodo } from '../../api/todos';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  filter: Filter;
+  setFilter: (filter: Filter) => void;
+  setTodos: (todos: Todo[]) => void;
+  filteredTodos: Todo[];
+  todos: Todo[];
+  setErrors: (errors: Errors) => void;
+  setDeleting: (id: number[]) => void;
+};
+
+export const Footer: FC<Props> = ({
+  filter,
+  setFilter,
+  setTodos,
+  filteredTodos,
+  todos,
+  setErrors,
+  setDeleting,
+}) => {
+  const clearCompleted = async () => {
+    let completed = filteredTodos.filter((todo) => todo.completed);
+
+    completed = filteredTodos.filter((todo) => todo.completed);
+
+    const completedIds = completed.map((todo) => todo.id!);
```

Comment:
don't do like that
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/91411692/c8bfa6d4-47d3-4253-b7c7-7a1efeddc080)

```suggestion
    const completedIds = completed.map((todo) => todo.id);
```